### ISSUE TITLE
[FEATURE] suggest catalogItems_2020-12-01.json change attributes type to object

### ISSUE THREAD

**Issue by ihoment-meihui** [amzn/selling-partner-api-models#195]
The Java SDK for the CatalogItem API using Swagger Code Generator cannot get (parse) attributes from SP-API response.

**URL of the content on GitHub**
https://github.com/amzn/selling-partner-api-models/blob/main/models/catalog-items-api-model/catalogItems_2020-12-01.json

**Description of what the documentation currently says**
    "Item": {
      "description": "An item in the Amazon catalog.",
      "properties": {
        "asin": {
          "$ref": "#/definitions/ItemAsin"
        },
        "attributes": {
          "$ref": "#/definitions/ItemAttributes"
        },
       ...

**Description of what the documentation should say**
    "Item": {
      "description": "An item in the Amazon catalog.",
      "properties": {
        "asin": {
          "$ref": "#/definitions/ItemAsin"
        },
        "attributes": {
            "description": "A JSON object that contains structured item attribute data keyed by attribute name. Catalog item attributes are available only to brand owners and conform to the related product type definitions available in the Selling Partner API for Product Type Definitions.",
	    "additionalProperties": true,
	    "type": "object"
        },
       ...

**change attributes class to  Object**
![image](https://user-images.githubusercontent.com/35246538/145170412-e7a10256-688b-4f5d-8d12-a8eb8bc5cdb8.png)

**Comment by chapmanjw** [comment]
The `attributes` schema is a reference that already defines this as an object with additional properties: 

https://github.com/amzn/selling-partner-api-models/blob/main/models/catalog-items-api-model/catalogItems_2020-12-01.json#L1078-L1082

```
    "ItemAttributes": {
      "description": "A JSON object that contains structured item attribute data keyed by attribute name. Catalog item attributes are available only to brand owners and conform to the related product type definitions available in the Selling Partner API for Product Type Definitions.",
      "additionalProperties": true,
      "type": "object"
    },
```

Nesting that shouldn't have a material difference here.

Swagger codegen does not play well with Swagger 2.0 models and `additionalProperties`. For our own testing and such, we use the generated client to make the "raw" call and handle the JSON directly.

```
Call call = catalogApi.getCatalogItemCall(asin, marketplaces, includedDataList, null, null, null);
Response response = call.execute();
ResponseBody responseBody = response.body();
byte[] responseBytes = responseBody.bytes();
```

And then use your favorite JSON handler for reading the bytes as JSON (such as using the Jackson ObjectMapper to convert to JsonNode). 

I'll also share this Github issue with the team that handles the client generation code and sample to see if there is an alternative here.

**Comment by chapmanjw** [comment]
@ihoment-meihui Another option you can use here is to override the mapping of the `ItemAttributes` object to the generic JSON object of the serializer you are using with the Swagger codegen. 

For example, with GSON:

```
--import-mappings ItemAttributes=com.google.gson.JsonObject
```

Or with Jackson:

```
--import-mappings ItemAttributes=com.fasterxml.jackson.databind.node.ObjectNode
```

This results in the `attributes` property of the generated code using one of these generic JSON objects instead of the Java `Object` type.

**Comment by chapmanjw** [comment]
@ihoment-meihui Depending on how fancy you want to get with a solution here, you could also generate POJOs from the JSON Schema from the Product Type Definitions API and then use the Import Mappings to map the ItemAttributes to the POJOs generated from the JSON Schema.

**Comment by ihoment-meihui** [comment]
@chapmanjw Thanks. I tried with GSON

`--import-mappings ItemAttributes=com.google.gson.JsonObject`
, but the effect did not achieve what I wanted
Here is still ItemAttributes, not JSONObject.
![image](https://user-images.githubusercontent.com/35246538/146911455-97f3f9e9-bbd2-4d0c-9e52-00d21d33b0ed.png)

**Comment by ihoment-meihui** [comment]
I expect the code generated will look like
![image](https://user-images.githubusercontent.com/35246538/146911828-ea32e62e-3ced-42ec-91e4-e7096740e03c.png)
Or
![image](https://user-images.githubusercontent.com/35246538/146911942-8739a88f-c3ee-4280-a52f-18d3db45f615.png)

**Comment by ihoment-meihui** [comment]
I use the swagger-codegen-cli-2.4.13.jar downloaded from https://github.com/amzn/selling-partner-api-docs/blob/main/guides/en-US/developer-guide/SellingPartnerApiDeveloperGuide.md#generating-a-java-sdk-with-lwa-token-exchange-and-authentication.
After I replaced 'import-mapping' with 'type-mappings',
`--type-mappings ItemAttributes=com.google.gson.JsonObject`
I got the result I expected.

![image](https://user-images.githubusercontent.com/35246538/146916778-19802cf5-0e76-434b-9b49-f08c540cb58b.png)

Thanks for your advice.
